### PR TITLE
Retail predictions

### PR DESCRIPTION
**Pull request by a-sahakyan**

### CODE REVIEW

File: retail/interactive-tutorial/RetailPredictions.Samples/PredictionWithParametersSample.cs [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+namespace RetailPredictions.Samples
+{
+    /// <summary>
+    /// The prediction with parameters sample class.
+    /// </summary>
+    public class PredictionWithParametersSample
+    {
+        private const string Endpoint = "retail.googleapis.com";
+
+        private static readonly string ProjectNumber = Environment.GetEnvironmentVariable("PROJECT_NUMBER");
```
Comment (amanda-tarafa) [inline retail/interactive-tutorial/RetailPredictions.Samples/PredictionWithParametersSample.cs:34]: We are not doing this as part of the sample, please take a look at the rest of Restail sample that have been merged and follow those as a template.

**Review by amanda-tarafa — CHANGES_REQUESTED** [review]
I've only reviewed a couple of samples, but these are using the wrong structure.
Please taka a look at the samples we have already merged for Retail and use those as templates.
Also, remove the setup files from this PR.

**Comment by amanda-tarafa** [comment]
Also, some tests are failing:

```
Failed RetailPredictions.Samples.Tests.PredictionFilteringTest.TestPredictionSimple [507 ms]
  Error Message:
   Grpc.Core.RpcException : Status(StatusCode="NotFound", Detail="servingConfig 'projects/1061780273775/locations/global/catalogs/default_catalog/servingConfigs/prediction' not found for catalog projects/1061780273775/locations/global/catalogs/default_catalog and project 1061780273775. If you just created this resource (for example, by activating your project), it may take up 5 minutes for the resource(s) to be activated.", DebugException="Grpc.Core.Internal.CoreErrorDetailException: {"created":"@1650533092.656747293","description":"Error received from peer ipv4:74.125.195.95:443","file":"/var/local/git/grpc/src/core/lib/surface/call.cc","file_line":1067,"grpc_message":"servingConfig 'projects/1061780273775/locations/global/catalogs/default_catalog/servingConfigs/prediction' not found for catalog projects/1061780273775/locations/global/catalogs/default_catalog and project 1061780273775. If you just created this resource (for example, by activating your project), it may take up 5 minutes for the resource(s) to be activated.","grpc_status":5}")
```

File: retail/interactive-tutorial/RetailPredictions.Samples/setup/UpdateUserEventsJson.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
+﻿using System;
```
[the remark this answers is not in the document]
Comment (t-karasova) [reply retail/interactive-tutorial/RetailPredictions.Samples/setup/UpdateUserEventsJson.cs:1]: HI Amanda, the PR #1699 contains bash scripts to setup the gcloud environment, and perform initial data importing to the catalog. It does not have any changes in the code samples. I suggest to not move this classes to that PR

File: retail/interactive-tutorial/RetailPredictions.Samples/PredictionWithParametersSample.cs [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+using System.Linq;
+
+namespace RetailPredictions.Samples
+{
+    /// <summary>
+    /// The prediction with parameters sample class.
+    /// </summary>
+    public class PredictionWithParametersSample
+    {
+        private const string Endpoint = "retail.googleapis.com";
```
[the remark this answers is not in the document]
Comment (a-sahakyan) [reply retail/interactive-tutorial/RetailPredictions.Samples/PredictionWithParametersSample.cs]: I Updated this code

File: retail/interactive-tutorial/RetailPredictions.Samples/PredictionWithParametersSample.cs (hunk 1 above)
Comment (a-sahakyan) [reply retail/interactive-tutorial/RetailPredictions.Samples/PredictionWithParametersSample.cs]: I Updated this code

File: retail/interactive-tutorial/RetailPredictions.Samples.Tests/PredictionWithParametersTest.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+using Google.Cloud.Retail.V2;
+using Xunit;
+
+namespace RetailPredictions.Samples.Tests
+{
+    public class PredictionWithParametersTest
+    {
+        [Fact]
+        public void TestPredictionSimple()
```
Comment (amanda-tarafa) [inline retail/interactive-tutorial/RetailPredictions.Samples.Tests/PredictionWithParametersTest.cs:23]:
```suggestion
        public void TestPredictionWithParameters()
```

**Comment by amanda-tarafa** [comment]
And tests are failing.

**Comment by amanda-tarafa** [comment]
I'd prefer if we did as we've done for all the rest of samples, that is to have a PR with just the samples and the tests, and another one for setup and cleanup.
Unless there's something I'm missing about this that is different to the rest.

About wether we have one or many cleanup and setup PRs, less discuss that internally. I've sent an email a couple of days ago looking for clarification as we have many setup and cleanup PRs that seems to overlap some. Thanks!!

File: retail/interactive-tutorial/RetailPredictions.Samples/PredictionWithParametersSample.cs [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+            predictRequest.UserEvent.ProductDetails.Add(new ProductDetail
+            {
+                Product = new Product
+                {
+                    Id = "55106" // An id of real product
+                }
+            });
+
+            // Try to update `priceRerankLevel` here
```
[the remark this answers is not in the document]
Comment (a-sahakyan) [reply retail/interactive-tutorial/RetailPredictions.Samples/PredictionWithParametersSample.cs:72]: This line of code, the user can set different values to test how the price retaining level impact on the recommendations response. And in the tutorial we give the description of this feature, this comment just point on the request field

File: retail/interactive-tutorial/RetailPredictions.Samples/PredictionWithParametersSample.cs [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+            Value priceRerankLevel = new Value();
+            priceRerankLevel.StringValue = "low-price-reranking";
```
[the remark this answers is not in the document]
Comment (a-sahakyan) [reply retail/interactive-tutorial/RetailPredictions.Samples/PredictionWithParametersSample.cs:74]: I have updated this code

File: retail/interactive-tutorial/RetailPredictions.Samples/PredictionWithParametersSample.cs [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+            returnProduct.BoolValue = true;
+
+            Dictionary<string, Value> entries = new Dictionary<string, Value>()
+            {
+                { "returnProduct", returnProduct },
+                { "priceRerankLevel", priceRerankLevel },
+                { "diversityLevel", diversityLevel }
+            };
+
+            predictRequest.Params.Add(entries);
```
[the remark this answers is not in the document]
Comment (a-sahakyan) [reply retail/interactive-tutorial/RetailPredictions.Samples/PredictionWithParametersSample.cs:91]: I have updated this code

File: retail/interactive-tutorial/RetailPredictions.Samples/PredictionWithParametersSample.cs [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+        /// Call the Retail API to get prediction
+        /// </summary>
+        /// <param name="projectId">The current project id.</param>
+        /// <returns></returns>
+        public static PredictResponse GetPrediction(string projectId)
+        {
+            PredictRequest predictRequest = GetPredictRequest(projectId);
+            PredictResponse predictResponse = GetPredictionService().Predict(predictRequest);
+
+            if (predictResponse.Results is null || !predictResponse.Results.Any())
```
[the remark this answers is not in the document]
Comment (a-sahakyan) [reply retail/interactive-tutorial/RetailPredictions.Samples/PredictionWithParametersSample.cs:106]: I have updated this code

File: retail/interactive-tutorial/RetailPredictions.Samples/PredictionWithParametersSample.cs [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+            {
+                Console.WriteLine("The search operation returned no matching results.");
```
[the remark this answers is not in the document]
Comment (a-sahakyan) [reply retail/interactive-tutorial/RetailPredictions.Samples/PredictionWithParametersSample.cs:108]: I have updated this code

File: retail/interactive-tutorial/RetailPredictions.Samples.Tests/PredictionWithParametersTest.cs [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+        {
+            const int ExpectedResultCount = 20;
```
[the remark this answers is not in the document]
Comment (a-sahakyan) [reply retail/interactive-tutorial/RetailPredictions.Samples.Tests/PredictionWithParametersTest.cs:25]: I have updated this code

File: retail/interactive-tutorial/RetailPredictions.Samples.Tests/PredictionWithParametersTest.cs (hunk 4 above)
Comment (a-sahakyan) [reply retail/interactive-tutorial/RetailPredictions.Samples.Tests/PredictionWithParametersTest.cs:23]: I have updated this code

File: retail/interactive-tutorial/RetailPredictions.Samples/PredictionWithParametersSample.cs [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+    /// <summary>
+    /// The prediction with parameters sample class.
+    /// </summary>
+    public class PredictionWithParametersSample
+    {
+        /// <summary>
+        /// Get Prediction Service
+        /// </summary>
+        /// <returns></returns>
+        private static PredictionServiceClient GetPredictionService()
```
[the remark this answers is not in the document]
Comment (a-sahakyan) [reply retail/interactive-tutorial/RetailPredictions.Samples/PredictionWithParametersSample.cs:36]: I have updated this code

File: retail/interactive-tutorial/RetailPredictions.Samples/PredictionWithParametersSample.cs [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+                    VisitorId = "281639", // A unique identifier to track visitors
+                    EventTime = Timestamp.FromDateTime(DateTime.UtcNow)
+                }
+            };
+
+            predictRequest.UserEvent.ProductDetails.Add(new ProductDetail
+            {
+                Product = new Product
+                {
+                    Id = "55106" // An id of real product
```
[the remark this answers is not in the document]
Comment (a-sahakyan) [reply retail/interactive-tutorial/RetailPredictions.Samples/PredictionWithParametersSample.cs:68]: I have updated this code

File: retail/interactive-tutorial/RetailPredictions.Samples/PredictionWithParametersSample.cs [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+                Placement = defaultBranchName,
+                UserEvent = new UserEvent
+                {
+                    EventType = "detail-page-view",
+                    VisitorId = "281639", // A unique identifier to track visitors
+                    EventTime = Timestamp.FromDateTime(DateTime.UtcNow)
+                }
+            };
+
+            predictRequest.UserEvent.ProductDetails.Add(new ProductDetail
```
[the remark this answers is not in the document]
Comment (a-sahakyan) [reply retail/interactive-tutorial/RetailPredictions.Samples/PredictionWithParametersSample.cs:64]: I have updated this code

File: retail/interactive-tutorial/RetailPredictions.Samples/PredictionWithParametersSample.cs [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+            string defaultBranchName = $"projects/{projectId}/locations/global/catalogs/default_catalog/placements/prediction";
+
+            PredictRequest predictRequest = new PredictRequest
+            {
+                Placement = defaultBranchName,
+                UserEvent = new UserEvent
+                {
+                    EventType = "detail-page-view",
+                    VisitorId = "281639", // A unique identifier to track visitors
+                    EventTime = Timestamp.FromDateTime(DateTime.UtcNow)
```
[the remark this answers is not in the document]
Comment (a-sahakyan) [reply retail/interactive-tutorial/RetailPredictions.Samples/PredictionWithParametersSample.cs:60]: I have updated this code

**Comment by amanda-tarafa** [comment]
@mkoshurnikov let me know about this one

**Comment by mkoshurnikov** [comment]
Can be closed because it's outdated

**Pull request closed without merging**